src/ucp/core/ucp_ep.c

```diff
@@ -1153,13 +1166,62 @@ static void ucp_ep_discard_lanes_callback(void *request, ucs_status_t status,
     ucs_free(arg);
 }
 
+static void ucp_ep_discard_lanes(ucp_ep_h ep, ucs_status_t discard_status)
```

Comment:
why we still need to purge from ucp_ep_discard_lanes_callback?
can't we remove it altogether ?

Reply:
get rid of it the 3rd commit, but it requires some test fixes